src/main/java/com/artipie/scheduling/ScriptScheduler.java

```diff
@@ -0,0 +1,134 @@
+     * @param clazz Class of the Job.
+     * @param data Map Data for the job's JobDataMap.
+     * @param <T> Class type parameter.
+     */
+    public <T extends Job> void scheduleJob(
+        final String cronexp, final Class<T> clazz, final Map<String, Object> data
+    ) {
+        try {
+            this.service.schedulePeriodicJob(cronexp, clazz, new JobDataMap(data));
+        } catch (final SchedulerException exc) {
+            throw new ArtipieException(exc);
+        }
+    }
+
+    /**
+     * Loads crontab from settings.
+     * Format is:
+     * <pre>
+     *     meta:
+     *       crontab:
+     *         - key: scripts/script1.groovy
+     *           cronexp: * * 10 * * ?
+     *         - key: scripts/script2.groovy
+     *           cronexp: * * 11 * * ?
+     * </pre>
+     * @param settings Artipie settings
+     */
+    @SuppressWarnings("PMD.AvoidDuplicateLiterals")
+    public void loadCrontab(final Settings settings) {
+        final CronDefinition crondef =
+            CronDefinitionBuilder.instanceDefinitionFor(CronType.QUARTZ);
+        final CronParser parser = new CronParser(crondef);
+        settings.crontab()
+            .map(
+                crontab ->
+                    crontab.values().stream()
+                        .map(YamlNode::asMapping)
+                        .map(
+                            yaml -> {
+                                final String key = yaml.string("key");
+                                final String cronexp = yaml.string("cronexp");
+                                boolean valid = false;
+                                try {
+                                    parser.parse(cronexp).validate();
+                                    valid = true;
+                                } catch (final IllegalArgumentException exc) {
+                                    Logger.error(
+                                        ScriptScheduler.class,
+                                        "Invalid cron expression %s %[exception]s",
+                                        cronexp,
+                                        exc
+                                    );
+                                }
+                                if (valid) {
+                                    final JobDataMap data = new JobDataMap();
+                                    data.put("key", key);
+                                    data.put("settings", settings);
+                                    try {
+                                        this.service.schedulePeriodicJob(
+                                            cronexp, ScriptRunner.class, data
+                                        );
+                                    } catch (final SchedulerException ex) {
+                                        throw new ArtipieException(ex);
+                                    }
+                                }
+                                return null;
```

Comment:
imho, this id a bad solution. Think, we should filter the stream values.

Reply:
Do you mean I should switch to something like `forEach()` instead of `map()`?

Reply:
Yes, we don't need to collect nulls.